src/Query/TimeoutExecutor.php

```diff
@@ -12,7 +12,7 @@ final class TimeoutExecutor implements ExecutorInterface
     private $loop;
     private $timeout;
 
-    public function __construct(ExecutorInterface $executor, $timeout, LoopInterface $loop = null)
+    public function __construct(ExecutorInterface $executor, float $timeout, ?LoopInterface $loop = null)
```

Comment:
Suggested changes LGTM, but there's a bit of overlap with the type additions in #225, so I would suggest keeping it at this for this PR:

```suggestion
    public function __construct(ExecutorInterface $executor, $timeout, ?LoopInterface $loop = null)
```

Reply:
Good catch :+1: